## UN Data Exploration

## #3

In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## #4

In [7]:
gdp_df = pd.read_csv("../data/gpd_per_capita.csv")

In [9]:
print(gdp_df)

     Country or Area  Year        Value  Value Footnotes
0        Afghanistan  2021  1517.016266              NaN
1        Afghanistan  2020  1968.341002              NaN
2        Afghanistan  2019  2079.921861              NaN
3        Afghanistan  2018  2060.698973              NaN
4        Afghanistan  2017  2096.093111              NaN
...              ...   ...          ...              ...
7657        Zimbabwe  1994  2670.106615              NaN
7658        Zimbabwe  1993  2458.783255              NaN
7659        Zimbabwe  1992  2468.278257              NaN
7660        Zimbabwe  1991  2781.787843              NaN
7661        Zimbabwe  1990  2704.757299              NaN

[7662 rows x 4 columns]


In [10]:
gdp_df.head(10)


Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2021  1517.016266              NaN
1     Afghanistan  2020  1968.341002              NaN
2     Afghanistan  2019  2079.921861              NaN
3     Afghanistan  2018  2060.698973              NaN
4     Afghanistan  2017  2096.093111              NaN
5     Afghanistan  2016  2101.422187              NaN
6     Afghanistan  2015  2108.714173              NaN
7     Afghanistan  2014  2144.449634              NaN
8     Afghanistan  2013  2165.340915              NaN
9     Afghanistan  2012  2122.830759              NaN

In [15]:
gdp_df.footer(10)


Country or Area  Year        Value  Value Footnotes
7652        Zimbabwe  1999  2866.032886              NaN
7653        Zimbabwe  1998  2931.725144              NaN
7654        Zimbabwe  1997  2896.147308              NaN
7655        Zimbabwe  1996  2867.026043              NaN
7656        Zimbabwe  1995  2641.378271              NaN
7657        Zimbabwe  1994  2670.106615              NaN
7658        Zimbabwe  1993  2458.783255              NaN
7659        Zimbabwe  1992  2468.278257              NaN
7660        Zimbabwe  1991  2781.787843              NaN
7661        Zimbabwe  1990  2704.757299              NaN

## #5

In [19]:
gdp_df = gdp_df.drop("Value Footnotes", axis = 1)

In [22]:
gdp_df = gdp_df.rename(columns = {"Country or Area": "Country", "Year":"Year", "Value" : "GDP_Per_Capita"})

In [23]:
gdp_df

Country  Year  GDP_Per_Capita
0     Afghanistan  2021     1517.016266
1     Afghanistan  2020     1968.341002
2     Afghanistan  2019     2079.921861
3     Afghanistan  2018     2060.698973
4     Afghanistan  2017     2096.093111
...           ...   ...             ...
7657     Zimbabwe  1994     2670.106615
7658     Zimbabwe  1993     2458.783255
7659     Zimbabwe  1992     2468.278257
7660     Zimbabwe  1991     2781.787843
7661     Zimbabwe  1990     2704.757299

[7662 rows x 3 columns]

## #6

In [26]:
gdp_df.shape

(7662, 3)

In [27]:
gdp_df.dtypes

Country            object
Year                int64
GDP_Per_Capita    float64
dtype: object

### There are 7662 rows and 3 columns. The data types are expected.

## #7

In [28]:
gdp_df["Year"].count()

7662

In [34]:
Years = gdp_df["Year"].value_counts()
print(Years)

Year
2013    242
2016    242
2014    242
2015    242
2020    242
2017    242
2018    242
2019    242
2021    241
2012    240
2011    240
2010    239
2009    239
2008    238
2007    237
2006    237
2004    236
2005    236
2003    235
2002    235
2001    234
2000    233
2022    232
1999    227
1998    226
1997    226
1996    223
1995    223
1994    213
1993    211
1992    210
1991    208
1990    207
Name: count, dtype: int64


In [41]:
Years = Years.sort_values(axis = 0, ascending = False)
print(Years)

Year
2013    242
2016    242
2019    242
2018    242
2020    242
2017    242
2015    242
2014    242
2021    241
2012    240
2011    240
2010    239
2009    239
2008    238
2007    237
2006    237
2005    236
2004    236
2003    235
2002    235
2001    234
2000    233
2022    232
1999    227
1998    226
1997    226
1996    223
1995    223
1994    213
1993    211
1992    210
1991    208
1990    207
Name: count, dtype: int64


### Years from 1990 - 2022 are included, and it seems to pretty consistently go up throughout the years excluding 2021 and 2022, so some countries didn't give their data for those two previous years.

## #8

In [42]:
gdp_df["Country"].value_counts()

Country
Least developed countries: UN classification    33
Middle East & North Africa                      33
Middle East & North Africa (IDA & IBRD)         33
Middle income                                   33
Mongolia                                        33
                                                ..
Kosovo                                          15
Sint Maarten (Dutch part)                       14
Turks and Caicos Islands                        12
Somalia                                         10
Djibouti                                        10
Name: count, Length: 242, dtype: int64

### There are 242 different countries represented in the dataset, which seems to show that it is not only countries represented in the listing. 

### Those countries at the bottom might have less obvservations because they are newly recognized countries potentially.